### PR TITLE
MOD-6597: Fix numeric index deletion

### PR DESCRIPTION
**Pull request by raz-mon**
**Describe the changes in the pull request**

We have a bug in our numeric index - we don't update the `revisionId` when we trim the Tree (in `NumericRangeTree_TrimEmptyLeaves`), causing us not to abort the reading of the numeric index for an invalidated cursor upon opening an invalidated `NumericRangeIterator`, and possibly crash.

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: tests/pytests/test_cursors.py [hunk 1]
```diff
@@ -416,3 +416,43 @@ def testCursorDepletionStrictTimeoutPolicy():
...
+    num_docs = 101
+    for i in range(num_docs):
+        conn.execute_command('hset', f'doc{i}', 'test', str(i))
+
+    # Initialize a cursor
+    res, cid = env.execute_command('ft.aggregate', 'idx', f'@test:[1 {num_docs}]', 'LOAD', '1', '@test', 'WITHCURSOR', 'COUNT', '1')
+    n = len(res) - 1
+
+    # Make sure GC is not self-invoked (periodic run).
+    env.expect('FT.CONFIG', 'SET', 'FORK_GC_RUN_INTERVAL', 3600).equal('OK')
```
Comment (oshadmi) [inline tests/pytests/test_cursors.py:441]: Consider using `num_docs` to be less than `FORK_GC_CLEAN_THRESHOLD` so it is stable

Comment (raz-mon) [reply tests/pytests/test_cursors.py:441]: Not sure I understand your meaning.
If it is set to less than `FORK_GC_CLEAN_THRESHOLD` the GC won't clean the index.

File: src/numeric_index.c [hunk 2]
```diff
@@ -516,6 +516,12 @@ NRN_AddRv NumericRangeTree_TrimEmptyLeaves(NumericRangeTree *t) {
   NRN_AddRv rv = {.numRanges = 0,
                   .changed = 0 };
   NumericRangeNode_RemoveChild(&t->root, &rv);
+  if (rv.changed) {
+    // Update the NumericTree
+    t->revisionId++;
+    t->numRanges += rv.numRanges;
+    t->emptyLeaves = 0;
```
[the remark this answers is not in the document]
Comment (raz-mon) [reply src/numeric_index.c:523]: Confirmed.
`rv.changed` is set once there was **any** change in the structure of the Tree.
If there wasn't any change, there is no need to update these fields.

[the remark this answers is not in the document]
Comment (oshadmi) [reply src/numeric_index.c:523]: Not in this PR: need to review when `_NUMERIC_RANGES_PARENTS` > 0
that `emptyLeaves` could remain > 0.

File: tests/pytests/test_cursors.py (hunk 1 above)
Comment (oshadmi) [reply tests/pytests/test_cursors.py:441]: Sorry, I meant equals not less than.

Comment (raz-mon) [reply tests/pytests/test_cursors.py:442]: Added 👍

**Review by oshadmi — APPROVED** [review]
👍🏼 👏🏼

**Pull request merged**